core/src/test/scala/integration/kafka/server/KRaftClusterTest.scala

```diff
@@ -792,6 +792,43 @@ class KRaftClusterTest {
     }
   }
 
+  /**
+   * Test that setting the Confluent-specific configuration
+   * confluent.apply.create.topic.policy.to.create.partitions has the expected effect.
+   */
+  @ParameterizedTest
+  @ValueSource(strings = Array("3.7-IV0", "3.7-IV2"))
+  def testCreatePartitions(metadataVersionString: String): Unit = {
```

Comment:
just curios. Is this test related to the constructor of PartitionAssignment?

Reply:
This test passes before and after the change. I just added it because I noticed a test gap in this area.